app/stores/ada/TrezorSendStore.js

```diff
@@ -0,0 +1,177 @@
+  },
+  connectError103: {
+    id: 'wallet.trezor.dialog.step.connect.error.103',
+    defaultMessage: '!!!Cancelled. Please retry.',
+    description: '<Cancelled. Please retry.> on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  saveError101: {
+    id: 'wallet.trezor.dialog.step.save.error.101',
+    defaultMessage: '!!!Falied to save. Please check your Internet connection and retry.',
+    description: '<Falied to save. Please check your Internet connection and retry.> on the Connect to Trezor Hardware Wallet dialog.'
+  },
+  error101: {
+    id: 'wallet.send.trezor.error.101',
+    defaultMessage: '!!!Signing cancelled on Trezor device. Please retry.',
+    description: '<Signing cancelled on Trezor device. Please retry.> on the Trezor send ADA confirmation dialog.'
+  },  
+});
+
+/** Note: Handles Trezor Signing */
+export default class TrezorSendAdaStore extends Store {
+  // =================== VIEW RELATED =================== //
+  @observable isActionProcessing: boolean = false;
+  @observable error: ?LocalizableError;
+  // =================== VIEW RELATED =================== //
+
+  // =================== API RELATED =================== //
+  createTrezorSignTxDataRequest: LocalizedRequest<CreateTrezorSignTxDataResponse> =
+    new LocalizedRequest(this.api.ada.createTrezorSignTxData);
+  // =================== API RELATED =================== //
+
+  setup() {
+    const trezorSendAction = this.actions.ada.trezorSend;
+    trezorSendAction.sendUsingTrezor.listen(this._sendUsingTrezor);
+    trezorSendAction.cancel.listen(this._cancel);
+  }
+
+  // TODO: [TREZOR] clear error state on initialization
+
+  /** Generates a payload with Trezor format and tries Trezor signing */
+  _sendUsingTrezor = async (params: CreateTrezorSignTxDataRequest): Promise<void> => {
+    // TODO: [TREZOR] fix type if possible
+    let trezorResp: any;
+    try {
+      this._setError(null);
+      this._setActionProcessing(true);
+
+      const { wallets, addresses } = this.stores.substores[environment.API];
+      const activeWallet = wallets.active;
+      if (!activeWallet) {
+        throw new Error('Active wallet required before sending.');
+      }
+      const accountId = addresses._getAccountIdByWalletId(activeWallet.id);
+      if (!accountId) {
+        throw new Error('Active account required before sending.');
+      }
+
+      this.createTrezorSignTxDataRequest.reset();
+
+      const {
+        trezorSignTxPayload,
+        changeAddress
+      } = await this.createTrezorSignTxDataRequest.execute(params).promise;
+  
+      trezorResp = await TrezorConnect.cardanoSignTransaction({ ...trezorSignTxPayload });
+      if (trezorResp && trezorResp.success) {
+        // TODO: [TREZOR] fix type if possible
+        const payload: any = trezorResp.payload;
+        // TODO: [TREZOR] use LocalizedRequest for any API call
+        await this.api.ada.sendHardwareTransaction({
```

Comment:
@nicarq @SebastienGllmt 

should we change this direct API call to `LocalizedRequest` ?

Reply:
Yes, please!